File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        seen = {}

        for i in range(n):
            rem = target - nums[i]
            if rem in seen:
                return [seen[rem],i]
            else:
                seen[nums[i]] = i
        
        

File: 0001-two-sum/0001-two-sum.cpp
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> seen;

        for (int i = 0; i < nums.size(); i++) {
            int rem = target - nums[i];

            if (seen.count(rem)) {
                return { seen[rem], i };
            }

            seen[nums[i]] = i;
        }

        return {};
    }
};
